### DISCUSSION TITLE
gatsby-plugin-image: resize images taking aspect ratio into account

### DISCUSSION THREAD

**Discussion by smashercosmo** [gatsbyjs/gatsby#30572] [Ideas / Feature Requests]
Currently, if you provide both width and height to `gatsbyImageData` function you get square image in response for both portrait and landscape oriented images:
```graphql
image {
  childImageSharp {
    gatsbyImageData(width: 450, height: 450)
  }
}
```

What I suggest is the ability to resize images according to this formula:
```js
const ratio = Math.min(width / imageWidth, height / imageHeight)
const resizedWidth =  imageWidth * ratio
const resizedHeight = imageHeight * ratio
```

Using this formula resized images will keep their orientation and aspect ratio. WDYT?

**Comment by smashercosmo** [comment]
@wardpeet wdyt?

**Reply by smashercosmo** [reply]
Anyone? I think it's pretty important feature to have

**Comment by ascorbic** [comment]
@smashercosmo  You probably want `transformOptions: {fit: CONTAIN}`. Take a look at [the sharp docs](https://sharp.pixelplumbing.com/api-resize) for the various options there.

**Reply by smashercosmo** [reply]
@ascorbic hey, thx for the tip. I tried all of the options. Unfortunately none of the options worked for me.
COVER, CONTAIN and FILL options behave the same way, when both width and height are provided (squared images are returned).

width INSIDE option, image is resized liked this:
```
{ "height": 3216,"width": 4284 }
              |
              V
{"height": 338, "width": 450 }
```

which looks correct, but another image (where height is bigger) is resized incorrectly
```
{ "height": 4032,"width": 3024 }
              |
              V
{"height": 599.112426035503, "width": 450 }
```

OUTSIDE option doesn't work the way I need either

**Reply by smashercosmo** [reply]
There is also a chance that it's a bug in Sharp

**Reply by smashercosmo** [reply]
@ascorbic WDYT?

**Comment by smashercosmo** [comment]
Still relevant for Gatsby 3.3

**Comment by smashercosmo** [comment]
Still not possible to resize an image so that it's smaller than or equal to specified dimensions while maintaining aspect ratio :(

**Reply by smashercosmo** [reply]
ping @ascorbic

**Reply by ascorbic** [reply]
I don't work at Gatsby anymore I'm afraid and don't really have time to look at this.

**Reply by smashercosmo** [reply]
alright, thx, I'll try to ping someone else

**Reply by smashercosmo** [reply]
@LekoArts @wardpeet may be some of you cam comment on this?
